docs: Update breaking changes on webFrame.setSpellCheckProvider (#17915)
electron/electron: docs/api/breaking-changes.md

  })
```

## `webFrame.setSpellCheckProvider`

> The `spellCheck` callback is now asynchronous, and `autoCorrectWord` parameter does not exists anymore.

Reviewer: ```suggestion
The `spellCheck` callback is now asynchronous, and `autoCorrectWord` parameter has been removed.
```

```js
// Deprecated
webFrame.setSpellCheckProvider('en-US', true, {
  spellCheck: (text) => {
    return !spellchecker.isMisspelled(text)
  }
})
// Replace with
webFrame.setSpellCheckProvider('en-US', {
  spellCheck: (words, callback) => {
    callback(words.filter(text => spellchecker.isMisspelled(text)))
  }
})

> ```

Reviewer: Thanks for adding this example!
